C23_WMDE_Mobile_DE_02

Sperling-0 opened the pull request:
- Uses CTRL from test 01
- CTRL has a text change and close button fix for full page banner
- VAR is a text change, headline change and close button fix for full page banner
- 'banner closed' cookie is not set when closing the fullpage banner for CTRL and VAR

Ticket: https://phabricator.wikimedia.org/T342204

Sperling-0 committed:
Archive C23_WMDE_Mobile_DE_01

Ticket: https://phabricator.wikimedia.org/T342204

Sperling-0 committed:
Prepare campaign for C23_WMDE_Mobile_DE_02

CTRL - Make a text change
CTRL - Fix fullpage banner close button
CTRL - 'banner closed' cookie is not set when closing the fullpage banner

Ticket: https://phabricator.wikimedia.org/T342204

Sperling-0 committed:
Fix fullpage banner close button for wpde_mobile

Fix it here as it is using the same theme - Mikings

Ticket: https://phabricator.wikimedia.org/T342204

Sperling-0 committed:
Implement VAR for C23_WMDE_Mobile_DE_02

VAR - Make a text
VAR - Fix fullpage banner close button
VAR - 'banner closed' cookie is not set when closing the fullpage banner

Ticket: https://phabricator.wikimedia.org/T342204

Abban commented:
@Sperling-0 re "The 'banner closed' cookie shouldn't be set when closing the large modal banner."

If you look in the `BannerCtrl.vue` and `BannerVar.vue` you can see the `FullPageBanner` close event is firing with `CloseChoices.Close`. This can be fired with `CloseChoices.MaybeLater` and it'll then not set the banner closed cookie.

Abban reviewed (changes requested):
Please add the feature tests for the VAR banner

Sperling-0 committed:
Fix tests for mobile banners

- The 'CloseChoice' for fullpage banner close button is 'MaybeLater'

Ticket: https://phabricator.wikimedia.org/T342204

Abban committed:
Style mobile full page close buttons

The mobile full pages are now modals meaning the
close button should be absolutely positioned.

This also adds a background to the buttons so they
look okay when they cover other elements.
